mlir/tools/rocmlir-lib/CMakeLists.txt

```diff
@@ -45,57 +47,344 @@ if(BUILD_FAT_LIBROCKCOMPILER)
   set(PACKAGE_NAME ${CMAKE_PROJECT_NAME_LOWER})
 
   set(LIBRARY_NAME rockCompiler)
-  function(combine_archives output_archive)
-    set(mri_file ${CMAKE_CURRENT_BINARY_DIR}/${output_archive}.mri)
-    set(full_output_path ${ROCMLIR_LIB_DIR}/lib${output_archive}.a)
-    set(output_archive_dummy_file ./${output_archive}.dummy.cpp)
+  set(COMPONENT_NAME ${LIBRARY_NAME})
+
+  set(__rocmlir_targets
```

Comment:
Reject for please find a way to not hard-code this. This'll be failure-prone and it's unclear why we needed it.

Reply:
While working on the Windows port, we discovered `file(GLOB..)` pulls more than required, polluting the fat library with unwanted `*.lib` files.

Reply:
Also, as a note, we've started a sync up with upstream, so the changes you had to make to `external/` shouldn't be needed anymore ... and will give you a pile of merge conflicts.

Second, please explain more about how those files were unwanted? And can you instead *remove* them from the list (perhaps with some more regex matching on the list element) instead of doing this, which, again, looks like something that'll get tedious to maintain and/or blow up in our face at the worst possible moment.